files/tk_metrics

```diff
@@ -7,44 +7,71 @@ require 'time'
 require 'optparse'
 require 'yaml'
 
-options = {}
+#===========================================================================#
+# BUILD CONFIGURATION OPTIONS                                               #
+#===========================================================================#
+
+# Define the list of mandatory options that must be specified on the CLI
+REQUIRED_FLAGS = {
+  metrics_type: { type: :string },
+}
+
+# Define the list of options that will be read from the config file, and which
+# may optionally be specified on the CLI.
+SETTINGS = {
+  output_dir: { type: :string  },
+  hosts:      { type: :list    },
+  port:       { type: :string  },
```

Comment:
I'm not 100% convinced that metrics will always run on the same port as the status endpoint so I don't see a need to rename this to port.  

If we decide to allow running the status endpoint on a different port so that even when a component is overloaded then I'd like to be able to have metrics_port and status_port.